platform/ios/src/MGLAnnotationView.mm

```diff
@@ -95,6 +103,129 @@ - (void)updateScaleForPitch:(CGFloat)pitch
     }
 }
 
+- (void)setDraggable:(BOOL)draggable
+{
+    _draggable = draggable;
+    if (draggable) {
+        [self enableDrag];
+    } else {
+        [self disableDrag];
+    }
+}
+
+- (void)enableDrag
+{
+    if (!_longPressRecognizer) {
+        UILongPressGestureRecognizer *recognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handleLongPress:)];
+        recognizer.delegate = self;
+        [self addGestureRecognizer:recognizer];
+        _longPressRecognizer = recognizer;
+    }
+    
+    if (!_panGestureRecognizer) {
+        UIPanGestureRecognizer *recognizer = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(handlePan:)];
+        recognizer.delegate = self;
+        [self addGestureRecognizer:recognizer];
+        _panGestureRecognizer = recognizer;
+    }
+}
+
+- (void)disableDrag
+{
+    [self removeGestureRecognizer:_longPressRecognizer];
+    [self removeGestureRecognizer:_panGestureRecognizer];
+}
+
+- (void)handleLongPress:(UILongPressGestureRecognizer *)sender
+{
+    if (sender.state == UIGestureRecognizerStateBegan)
+    {
+        self.dragState = MGLAnnotationViewDragStateStarting;
+    }
+    else if (sender.state == UIGestureRecognizerStateChanged)
+    {
+        self.dragState = MGLAnnotationViewDragStateDragging;
+    }
+    else if (sender.state == UIGestureRecognizerStateCancelled) {
+        self.dragState = MGLAnnotationViewDragStateCanceling;
+    }
+    else if (sender.state == UIGestureRecognizerStateEnded)
+    {
+        self.dragState = MGLAnnotationViewDragStateEnding;
+    }
+}
+
+- (void)handlePan:(UIPanGestureRecognizer *)sender
+{
+    CGPoint center = [sender locationInView:sender.view.superview];
+    [self setCenter:center pitch:self.mapView.camera.pitch];
+    
+    if (sender.state == UIGestureRecognizerStateEnded) {
+        self.dragState = MGLAnnotationViewDragStateNone;
+    }
+}
+
+- (void)setDragState:(MGLAnnotationViewDragState)dragState
+{
+    [self setDragState:dragState animated:YES];
+}
+
+- (void)setDragState:(MGLAnnotationViewDragState)dragState animated:(BOOL)animated
+{
+    _dragState = dragState;
+    
+    switch (dragState) {
+        case MGLAnnotationViewDragStateNone:
+            break;
+        case MGLAnnotationViewDragStateStarting: {
+            if (animated)
+            {
+                [UIView animateWithDuration:.20 animations:^{
+                    self.transform = CGAffineTransformScale(CGAffineTransformIdentity, 2, 2);
```

Comment:
You are absolutely right @1ec5 the animation stuff should not be included in `MGLAnnotationView`